### PR TITLE
refactor(cli): remove TextDocument

### PR DESCRIPTION
**Pull request by kitsonk**
PR refactors out `TextDocument` and converts everything to strings.

~Still WIP as there are the utf16 reads from the file system that I need to fix.~

This was based on a discussion with @ry that we shouldn't leak this all over the code base, makes it really awkward to deal with code.

### CODE REVIEW

File: cli/file_fetcher.rs [hunk 1]
```diff
@@ -32,58 +31,9 @@ use std::str;
...
-}
-
-impl From<&str> for TextDocument {
-  fn from(s: &str) -> Self {
-    TextDocument::new(s.as_bytes().to_vec(), Option::<&str>::None)
-  }
-}
+// pub fn to_str(&self) -> Result<Cow<str>, std::io::Error> {
+//   text_encoding::convert_to_utf8(&self.bytes, &self.charset)
+// }
```
Comment (ry) [inline cli/file_fetcher.rs:36]: Remove

File: cli/tsc.rs [hunk 2]
```diff
@@ -862,14 +862,10 @@ impl SourceMapGetter for TsCompiler {
   fn get_source_line(&self, script_name: &str, line: usize) -> Option<String> {
     self
       .try_resolve_and_get_source_file(script_name)
-      .and_then(|out| {
-        out.source_code.to_str().ok().map(|v| {
-          // Do NOT use .lines(): it skips the terminating empty line.
-          // (due to internally using .split_terminator() instead of .split())
```
Comment (ry) [inline cli/tsc.rs:868]: Retain this comment?

**Review by ry — APPROVED** [review]
Very nice - LGTM

I have a couple minor comments...

File: cli/file_fetcher.rs (hunk 1 above)
Comment (kitsonk) [reply cli/file_fetcher.rs:36]: DOH! yeah... 🤦

**Pull request merged**